tools/module/generate.py

```diff
@@ -0,0 +1,191 @@
+    if os.path.exists(path):
+        sys.stderr.write("The path provided already exists.\n")
+        sys.stderr.write("Module generation aborted.\n")
+        sys.exit(1)
+
+    print("Module directory", module_path)
+    print("Creating directories")
+
+    # Create the required directories
+    os.makedirs(path)
+    print("\t", path)
+    os.makedirs(src_path)
+    print("\t", src_path)
+    os.makedirs(tests_path)
+    print("\t", tests_path)
+    os.makedirs(config_path)
+    print("\t", config_path)
+
+    # Split our provided path
+    parts = ["module"] + os.path.relpath(path, module_path).split(os.sep)
+
+    print("Generating files")
+
+    # Write all of our files
+    with open(os.path.join(path, "CMakeLists.txt"), "w") as output:
+        output.write(generate_cmake(parts))
+        print("\t", os.path.join(path, "CMakeLists.txt"))
+
+    with open(os.path.join(path, "README.md"), "w") as output:
+        output.write(generate_readme(parts))
+        print("\t", os.path.join(src_path, "README.md"))
+
+    with open(os.path.join(src_path, "{}.h".format(module_name)), "w") as output:
+        output.write(generate_header(parts))
+        print("\t", os.path.join(src_path, "{}.h".format(module_name)))
+
+    with open(os.path.join(src_path, "{}.cpp".format(module_name)), "w") as output:
+        output.write(generate_cpp(parts))
+        print("\t", os.path.join(src_path, "{}.cpp".format(module_name)))
+
+    with open(os.path.join(tests_path, "{}.cpp".format(module_name)), "w") as output:
+        output.write(generate_test(parts))
+        print("\t", os.path.join(tests_path, "{}.cpp".format(module_name)))
+
+    with open(os.path.join(config_path, "{}.yaml".format(module_name)), "a"):
+        print("\t", os.path.join(config_path, "{}.yaml".format(module_name)))
+
+
+def generate_cmake(parts):
+    return textwrap.dedent(
+        """\
+        # Build our NUClear module
+        NUCLEAR_MODULE()
+        """
+    )
+
+
+def generate_header(parts):
+    template = textwrap.dedent(
+        """\
+        #ifndef {define}
+        #define {define}
+
+        #include <nuclear>
+
+        {openNamespace}
+
+            class {className} : public NUClear::Reactor {{
+
+            public:
+                /// @brief Called by the powerplant to build and setup the {className} reactor.
+                explicit {className}(std::unique_ptr<NUClear::Environment> environment);
+            }};
+
+        {closeNamespace}
+
+        #endif  // {define}
+        """
+    )
+
+    return template.format(
+        define="{}_H".format("_".join([p.upper() for p in parts])),
+        className=parts[-1],
+        openNamespace="\n".join(["namespace {} {{".format(x) for x in parts[:-1]]),
+        closeNamespace="\n".join("}" * (len(parts) - 1)),
+    )
+
+
+def generate_cpp(parts):
+    template = textwrap.dedent(
+        """\
+        #include "{className}.h"
+
+        #include "extension/Configuration.h"
```

Comment:
This needs to be removed from `nuclear/tools/module/generate.py`

Reply:
Sure